crates/iota-types/src/transaction.rs

```diff
@@ -307,21 +335,36 @@ impl EndOfEpochTransactionKind {
         protocol_version: ProtocolVersion,
         storage_charge: u64,
         computation_charge: u64,
+        computation_charge_burned: u64,
         storage_rebate: u64,
         non_refundable_storage_fee: u64,
         epoch_start_timestamp_ms: u64,
         system_packages: Vec<(SequenceNumber, Vec<Vec<u8>>, Vec<ObjectID>)>,
     ) -> Self {
-        Self::ChangeEpoch(ChangeEpoch {
-            epoch: next_epoch,
-            protocol_version,
-            storage_charge,
-            computation_charge,
-            storage_rebate,
-            non_refundable_storage_fee,
-            epoch_start_timestamp_ms,
-            system_packages,
-        })
+        if protocol_version.as_u64() > 3 {
```

Comment:
I don't like this solution. Hardcoding protocol versions like this is very intransparent.
Better have 2 initializers and switch on the call-site depending on a protocol-config parameter.